maps/castle_wall.go

```diff
@@ -0,0 +1,642 @@
+package maps
+
+import (
+	"github.com/BattlesnakeOfficial/rules"
+)
+
+func init() {
+	globalRegistry.RegisterMap("hz_castle_wall", CastleWallMediumHazardsMap{})
+	globalRegistry.RegisterMap("hz_castle_wall_lg", CastleWallLargeHazardsMap{})
+	globalRegistry.RegisterMap("hz_castle_wall_xl", CastleWallExtraLargeHazardsMap{})
+}
+
+func setupCastleWallBoard(startingPositions [][]rules.Point, hazards []rules.Point, initialBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	rand := settings.GetRand(initialBoardState.Turn)
+
+	// place snakes
+	var snakes []rules.Point
+	// always support up to 8x snakes on all supported board sizes
+	snakes = append(snakes, startingPositions[0]...)
+	if len(initialBoardState.Snakes) >= 5 {
+		snakes = append(snakes, startingPositions[1]...)
+	}
+
+	// only support 8 or less snakes on boards smaller than XLarge
+	if (len(initialBoardState.Snakes) > 8) && (initialBoardState.Width < rules.BoardSizeXLarge) {
+		return rules.ErrorTooManySnakes
+	}
+
+	// only support 12 or less snakes for xlarge board sizes
+	if (initialBoardState.Width >= rules.BoardSizeXLarge) && (len(initialBoardState.Snakes) > 12) {
+		return rules.ErrorTooManySnakes
+	}
+
+	// add positions 9-12 for xlarge board size
+	if (initialBoardState.Width >= rules.BoardSizeXLarge) && (len(initialBoardState.Snakes) > 8) {
+		snakes = append(snakes, startingPositions[2]...)
+	}
+
+	rand.Shuffle(len(snakes), func(i int, j int) {
+		snakes[i], snakes[j] = snakes[j], snakes[i]
+	})
+
+	for index, snake := range initialBoardState.Snakes {
+		head := snakes[index]
+		editor.PlaceSnake(snake.ID, []rules.Point{head, head, head}, rules.SnakeMaxHealth)
+	}
+
+	for _, h := range hazards {
+		editor.AddHazard(h)
+	}
+	return nil
+}
+
+func updateCastleWallBoard(food []rules.Point, lastBoardState *rules.BoardState, settings rules.Settings, editor Editor) error {
+	// no food spawning for first 10 turns
+	if lastBoardState.Turn < 10 {
+		return nil
+	}
+
+	// max of 2 food on medium & large board
```

Comment:
Same concept as above, pass in the max food as a parameter since it seems to be linked to map size, and just do one if statement